File: Data Structures & Algorithms/anagram-groups/submission-0.py
from typing import List
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        groups = defaultdict(list)
        def make_key(sp:str)->tuple:
            count=[0]*26
            for ch in sp:
                count[ord(ch)-ord('a')]+=1
            return tuple(count)
        for s in strs:
            key = make_key(s)
            groups[key].append(s)
        return list(groups.values())

File: Data Structures & Algorithms/is-anagram/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        let m = new Map();
        for(const i of s){
            if(m.has(i)){
                m.set(i,m.get(i)+1);
            }
            else m.set(i,1);
        }
        for(const i of t){
            if(!m.has(i))return false;
            else{
                let count = m.get(i);
                m.set(i,count-1);
            }
        }
        for(const [x,y] of m){
            if(y!=0)return false;
        }
        return true;
    }
}
